src/panels/lovelace/hui-root.ts

```diff
@@ -569,7 +572,12 @@ class HUIRoot extends LitElement {
         newSelectView = views.findIndex(this._isVisible);
         navigate(
           this,
-          `${this.route!.prefix}/${views[newSelectView].path || newSelectView}`,
+          constructUrl(
+            `${this.route!.prefix}/${
+              views[newSelectView].path || newSelectView
+            }`,
+            addSearchParam({ ...(this.lovelace!.editMode && { edit: "1" }) })
+          ),
```

Comment:
We don't want overhead when not running in editmode, I suggest we create a function:

```ts
private _selectView(viewPath) {
	if (!this.lovelace.editMode) {
		navigate(this, `${this.route!.prefix}/${viewPath}`)
		return;
    }
	navigate(this, `${this.route!.prefix}/${viewPath}&${addSearchParam({ edit: "1" })}`)
}

```

Reply:
Only drawback I can see is that if the user has other search params in the URL (for whatever use-case) we would loose those when not in edit mode, but we do that today as well, so probably unlikely scenario?

Reply:
yes